### PR TITLE
[SYCL] Fix spriv-to-ir-wrapper invocation in clang-linker-wrapper

### PR DESCRIPTION
**Pull request by maksimsab**
Fix command line argument passed to "spirv-to-ir-wrapper" invocation in clang-linker-wrapper.
Also make clang-linker-wrapper print arguments quoted if needed.

### CODE REVIEW

**Comment by maksimsab** [comment]
Also this test fixes the following E2E tests in New Offloading Model
SYCL :: Basic/multisource_spv_obj.cpp​
SYCL :: Basic/spirv_device_obj_smoke.cpp​

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 1]
```diff
@@ -159,8 +159,11 @@ void printCommands(ArrayRef<StringRef> CmdArgs) {
     return;
 
   llvm::errs() << " \"" << CmdArgs.front() << "\" ";
-  for (auto IC = std::next(CmdArgs.begin()), IE = CmdArgs.end(); IC != IE; ++IC)
-    llvm::errs() << *IC << (std::next(IC) != IE ? " " : "\n");
+  for (auto IC = std::next(CmdArgs.begin()), IE = CmdArgs.end(); IC != IE;
+       ++IC) {
+    llvm::sys::printArg(llvm::errs(), *IC, false);
```
Comment (sarnex) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:164]: can we add an inline comment of what `false` is

**Review by sarnex — APPROVED** [review]
CmdArgs part LGTM, I don't understand the printing part so I'll leave that to others.

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp (hunk 1 above)
Comment (maksimsab) [reply clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp]: Sure, added.

**Comment by maksimsab** [comment]
The point of my change is that we pass the string with whitespaces like "--spirv-option1 --spirv-option2" as one command line argument. The issue before my change was that a printed command was unquoted and it differs from the perspective of the shell. More precisely, it starts being interpreted as two command line arguments instead of one.
My change adds quotation when it is needed.

**Review by mdtoguchi — APPROVED** [review]

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 2]
```diff
@@ -162,8 +162,11 @@ void printCommands(ArrayRef<StringRef> CmdArgs) {
     return;
 
   llvm::errs() << " \"" << CmdArgs.front() << "\" ";
-  for (auto IC = std::next(CmdArgs.begin()), IE = CmdArgs.end(); IC != IE; ++IC)
-    llvm::errs() << *IC << (std::next(IC) != IE ? " " : "\n");
+  for (auto IC = std::next(CmdArgs.begin()), IE = CmdArgs.end(); IC != IE;
```
Comment (asudarsa) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:165]: Do we really need this change? This introduces changes in the way commands are emitted here and that causes test failures. 
@maksimsab , why do we need this change? Thanks

Comment (maksimsab) [reply clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:165]: Without this change you get printed command line that is different from one that has actually been run.

**Comment by maksimsab** [comment]
Update: Removed `printArg` usage. I will have to get it back more accurate directly in llvm-project.

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

**Comment by maksimsab** [comment]
@intel/llvm-gatekeepers Can we merge it? Failure in e2e Windows Intel GEN12 is irrelevant.

**Comment by sarnex** [comment]
Usually I would ask we rerun win gen12 testing but since I know the code changes has no e2e test coverage anyway, i'll just merge it now

**Pull request merged**